src/operation.rs

```diff
@@ -53,6 +54,31 @@ impl Typed for OpResult {
     }
 }
 
+impl Printable for OpResult {
+    fn fmt(
+        &self,
+        ctx: &Context,
+        _state: &printable::State,
+        f: &mut std::fmt::Formatter<'_>,
+    ) -> std::fmt::Result {
+        write!(f, "{}", self.unique_name(ctx))
+    }
+}
+
+impl Printable for Vec<OpResult> {
```

Comment:
Since `OpResult` implements `Typed`, wouldn't  `impl<T: Typed> TypedPrinter for &Vec<T> {` that you've written already provide this?

Reply:
No. We do not want to print the types, but the names of `OpResult`.

I find `OpResult` and the `..._result()` methods somewhat tricky to use. Right now I'm trying to use `pliron` from an external crate and running into a few minor issues. One of them is printing the results (because we do not export all symbols). 

I'm not exactly sure yet what to do about it, but currently I'm overcoming the issue by creating a "view" that references `Operation`:

```
impl Operation {
    ...
    pub fn results(&self) -> Results<'_> {
        Results::new(&self.results)
    }
    ...
}

#[derive(Clone)]
pub struct Results<'a> {
    results: &'a [OpResult],
}

impl<'a> Results<'a> {
    fn new(results: &'a [OpResult]) -> Self {
        Self { results }
    }

    pub fn len(&self) -> usize {
        self.results.len()
    }

    pub fn is_empty(&self) -> bool {
        self.results.is_empty()
    }

    pub fn get(&self, idx: usize) -> Option<Value> {
        self.results.get(idx).map(|res| res.into())
    }

    pub fn iter(&self) -> impl Iterator<Item = Value> + '_ {
        self.results.iter().map(|res| res.into())
    }
}
```

I wonder if that approach would also help to "reduce" the public interface of `Operation`. WDYT?

Reply:
I see no reason for `OpResult` to be a public type. Results should be used as their `Value`s. So I'm guessing providing `get_results` or `get_operands` that both return an `Iterator<Item = Value>` would be useful? Then you could use `iter_with_sep` to print it?

Reply:
Yes, `get_results` and `get_operands` would be useful. If the methods return an iterator maybe call the `results_iter` and `operands_iter`? Not just for printing, but in general iterations via iterators feel more Rust like instead of using `get_num_result` and `get_result_mut`.

I was just wondering if we want to even given a more array like view on the results instead of `get_num_result` + `get_result(_mut)` + an iterator. In that case we would need something like `Result<'a>` as we can't return `&[OpResult]` as is.